src/interfaces.js

```diff
@@ -73,5 +73,46 @@ interface RouteConfig {
   */
   caseSensitive?: boolean;
 
+  /**
+  * Add to specify an activation strategy if the strategy is always the same and you don't want that
+  * to be in your view-model code. Available values are 'replace' and 'invoke-lifecycle'.
+  */
+  activationStrategy?: string;
+  
   [x: string]: any;
 }
+
+interface RoutableComponent {
+  /**
+  * Implement this hook if you want to control whether or not your view-model can be navigated to.
+  * Return a boolean value, a promise for a boolean value, or a navigation command.
+  */
+  canActivate?: (params: any, routeConfig: RouteConfig, navigationInstruction: NavigationInstruction) => boolean|Promise<boolean>|PromiseLike<boolean>|NavigationCommand;
```

Comment:
Looks like babel 6 also [supports type aliases](https://github.com/codemix/babel-plugin-typecheck/pull/43), like TypeScript, so the return type of canActivate and canDeactivate could be defined in one place and used for both methods, smth like I've done [here](https://gist.github.com/atsu85/2777454d6ee215a1b9d3911ce157828d#file-viewhooks-d-ts-L16)